core/modules/replica/DatabaseServicesMySQL.cc

```diff
@@ -521,114 +570,174 @@ void DatabaseServicesMySQL::saveReplicaInfoImpl(util::Lock const& lock,
+                   .atChunk(replica.chunk()) = &replica;
+    }
+
+    LOGS(_log, LOG_LVL_DEBUG, context << "old replicas group: 2");
+
+    // Find differences between the collections
+
+    LOGS(_log, LOG_LVL_DEBUG, context << "*** replicas compa: 1");
+
+    WorkerDatabaseChunkMap<ReplicaInfo const*> inBoth;
+    SemanticMaps::intersect(newReplicas,
+                            oldReplicas,
+                            inBoth);
+
+    WorkerDatabaseChunkMap<ReplicaInfo const*> inNewReplicasOnly;
+    WorkerDatabaseChunkMap<ReplicaInfo const*> inOldReplicasOnly;
+    SemanticMaps::diff2(newReplicas,
+                        oldReplicas,
+                        inNewReplicasOnly,
+                        inOldReplicasOnly);
+
+    LOGS(_log, LOG_LVL_DEBUG, context << "*** replicas compa: 2"
+         << " #both: " << SemanticMaps::count(inBoth)
+         << " #new: " << SemanticMaps::count(inNewReplicasOnly)
+         << " #old: " << SemanticMaps::count(inOldReplicasOnly));
+
+    // Eiminate outdated replicas
+
+    LOGS(_log, LOG_LVL_DEBUG, context << "old replicas remov: 1");
+    
+    for (auto&& worker: inOldReplicasOnly.workerNames()) {
+
+        auto const& databases = inOldReplicasOnly.worker(worker);
+        for (auto&& database: databases.databaseNames()) {
+
+            auto const& chunks = databases.database(database);
+            for (auto&& chunk: chunks.chunkNumbers()) {
+                deleteReplicaInfoImpl(lock, worker, database, chunk);
             }
         }
+    }
 
-    } else {
+    LOGS(_log, LOG_LVL_DEBUG, context << "old replicas remov: 2");
 
-        // Bulk delete if the input collection is empty or has no context
-        _conn->execute(
-            "DELETE FROM " + _conn->sqlId(   "replica") +
-            "  WHERE " +     _conn->sqlEqual("worker",   worker) +
-            "    AND " +     _conn->sqlEqual("database", database));
+    // Insert new replicas not present in the old collection
+
+    LOGS(_log, LOG_LVL_DEBUG, context << "new replicas inser: 1");
+
+    for (auto&& worker: inNewReplicasOnly.workerNames()) {
+
+        auto const& databases = inNewReplicasOnly.worker(worker);
+        for (auto&& database: databases.databaseNames()) {
+
+            auto const& chunks = databases.database(database);
+            for (auto&& chunk: chunks.chunkNumbers()) {
+                ReplicaInfo const* ptr = chunks.chunk(chunk);
+                saveReplicaInfoImpl(lock, *ptr);
+            }
+        }
     }
 
-    // Finally push new (or update existing) replicas info into the database
-    // (some of those replicas will be brand new, others - will need to be updated)
-    for (auto&& info: infoCollection) {
-        saveReplicaInfoImpl(lock, info);
+    LOGS(_log, LOG_LVL_DEBUG, context << "new replicas inser: 2");
+
+    // Deep comparision of the replicas in the intersect area to see
+    // wich of those need to be updated.
+
+    LOGS(_log, LOG_LVL_DEBUG, context << "old replicas updat: 1");
```

Comment:
so many strange abbreviations makes things hard to read. 

IMHO.

Reply:
These are just truncations not abbreviations.
I can refactor these messages if you feel it will help to improve the clarity of the code.

Reply:
Eliminated these messages.